### PR TITLE
implement Dock trait

### PR DESCRIPTION
**Pull request by Swoboderz**
implementing Dock Trait almost in accordance to #16470,
In order to fully implement it, I would have to change a lot of code, even outside Land.cs to get the required information to function as normal. suggestions for it required.

### CODE REVIEW

File: OpenRA.Mods.Common/Activities/Resupply.cs [hunk 1]
```diff
@@ -42,7 +42,7 @@ public class Resupply : Activity
 
 		public Resupply(Actor self, Actor host, WDist closeEnough, bool stayOnResupplier = false)
 		{
-			this.host = Target.FromActor(host);
+			this.host = Target.FromPos(host.CenterPosition + host.Trait<Dock>().Info.EnterOffset);
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Activities/Resupply.cs:45]: I think this should use `TraitOrDefault` and a null check.

File: OpenRA.Mods.Common/Traits/Dock.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        public readonly WVec EnterOffset = WVec.Zero;
+        [Desc("Cell offset where the actor enters the dock relative to top left cell of the dockable actor.")]
+        public readonly CVec EntranceCell = CVec.Zero;
+
+        public readonly int Facing = -1;
+
+		/*[FieldLoader.Require]
+		[Desc("DockingTypes (from the Dockers trait) that are able to dock this.")]
+        public readonly BitSet<DockingType> Types = default(BitSet<DockingType>);
+        */
```
Comment (Mailaender) [inline OpenRA.Mods.Common/Traits/Dock.cs:31]: Left over out-commented code.

**Comment by Mailaender** [comment]
This needs a rebase and an update rule.

**Comment by abcdefg30** [comment]
No activity since quite a while, so closing this to clean the PR queue up.

**Pull request closed without merging**